promotion/service.go

```diff
@@ -404,24 +404,24 @@ func (s *Service) RunNextDrainJob(ctx context.Context) (bool, error) {
 	return s.datastore.RunNextDrainJob(ctx, s)
 }
 
-// RunNextPromotionNoPublicKey takes the next job and completes it
-func (s *Service) RunNextPromotionNoPublicKey(ctx context.Context) (bool, error) {
+// RunNextPromotionMissingIssuer takes the next job and completes it
+func (s *Service) RunNextPromotionMissingIssuer(ctx context.Context) (bool, error) {
```

Comment:
I'm looking at the job, and it seems like this runs every 5 seconds?

How does the service guarantee that a duplicate service isn't running at the same time?

Reply:
Currently it doesn't. We're relying on the fact that any duplicate issuer creation attempts on the challenge bypass microservice return an error ( due to a failing uniqueness constraint when inserting the issuer into it's own db ).

Reply:
https://github.com/brave-intl/challenge-bypass-server/blob/master/migrations/0002_rust_crypto.up.sql#L4 is the uniqueness constraint which we will hit when we try to insert here https://github.com/brave-intl/challenge-bypass-server/blob/master/server/db.go#L147